### PR TITLE
Add missing type declarations

### PR DESCRIPTION
**Pull request by nwalters512**
I did this in response to seeing a bug in another PR that would have been caught if type declarations had been there.

We'll want to add `noImplicitAny: true` to our compiler config in the future to automatically catch these. At this point, there are too many errors for that to be feasible.

### CODE REVIEW

**Review by SethPoulsen — APPROVED** [review]
The day we can add `noImplicitAny` will be awesome.

**Review by jonatanschroeder — CHANGES_REQUESTED** [review]
This still requires some work to handle the errors in typecheck. I'm marking as request changes to make it clear it still needs to be reviewed.

File: apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.js [hunk 1]
```diff
@@ -308,7 +308,7 @@ router.get(
         '',
       );
       const prefix = `${dirname}/`;
-      archive.append(null, { name: prefix });
+      archive.append('', { name: prefix });
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.js:311]: I just checked, and this call is unnecessary. Having it means the zip file will have an entry for the directory, but [the ZIP format](https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT) does not require separate entries for directories, unless some metadata is expected to be assigned to them (not the case here). Also, other directories created below based on paths that happen to contain a slash (e.g., if the QID has a path) don't have corresponding entries, and it still work. So the code here is correct, but could be removed altogether. Up to you.

**Review by jonatanschroeder — APPROVED** [review]

File: apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.js (hunk 1 above)
Comment (nwalters512) [reply apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.js:311]: I think this is here to ensure that even if there aren't any files, the folder still appears when unzipping. I think that's a useful property to maintain for consistency?

Comment (jonatanschroeder) [reply apps/prairielearn/src/pages/instructorAssessmentDownloads/instructorAssessmentDownloads.js:311]: Good point.

**Pull request merged**